pymc/model.py

```diff
@@ -920,111 +927,112 @@ def test_point(self) -> Dict[str, np.ndarray]:
-            Fallback option if initval is None and random draws are not implemented.
-            This is relevant for pm.Flat or pm.HalfFlat distributions and is subject
-            to ongoing refactoring of the initval API.
-        transform : optional, Transform
-            A transformation associated with the random variable.
-            Transformations are automatically applied to initial values.
+        jitter_rvs : set
+            The set (or list or tuple) of random variables for which a U(-1, +1) jitter should be
+            added to the initial value. Only available for variables that have a transform or real-valued support.
+        default_strategy : str
+            Which of { "moment", "prior" } to prefer if the initval setting for an RV is None.
+        overrides : dict
+            Initial value (strategies) to use instead of what's specified in `Model.initial_values`.
+        return_transformed : bool
+            If `True` the returned variables will correspond to transformed initial values.
 
         Returns
         -------
-        initval : np.ndarray
-            Numeric (transformed) initial value.
+        initial_point : dict
+            Maps transformed free variable names to transformed, numeric initial values.
         """
-        mode = get_mode(None)
-        opt_qry = mode.provided_optimizer.excluding("random_make_inplace")
-        mode = Mode(linker=mode.linker, optimizer=opt_qry)
 
-        if transform:
-            if initval is not None:
-                value = initval
-            else:
-                value = rv_var
-            rv_var = at.as_tensor_variable(transform.forward(rv_var, value))
+        def find_rng_nodes(variables):
+            return [
+                node
+                for node in aesara.graph.basic.graph_inputs(graph.outputs)
+                if isinstance(node, aesara.tensor.random.var.RandomStateSharedVariable)
+            ]
 
-        def initval_to_rvval(value_var, value):
-            rv_var = self.values_to_rvs[value_var]
-            initval = value_var.type.make_constant(value)
-            transform = getattr(value_var.tag, "transform", None)
-            if transform:
-                return transform.backward(rv_var, initval)
+        initial_values = make_initial_point_expression(
+            free_rvs=self.free_RVs,
+            rvs_to_values=self.rvs_to_values,
+            initval_strategies={**self.initial_values, **(overrides or {})},
+            jitter_rvs=jitter_rvs,
+            default_strategy=default_strategy,
+            return_transformed=return_transformed,
+        )
+        graph = FunctionGraph(outputs=initial_values, clone=False)
+
+        rng_nodes = find_rng_nodes(graph.outputs)
+        new_rng = [aesara.shared(np.random.RandomState(np.random.PCG64())) for _ in rng_nodes]
+        graph.replace_all(zip(rng_nodes, new_rng), import_missing=True)
+        func = compile_rv_inplace(inputs=[], outputs=graph.outputs)
```

Comment:
Should we use fast compile mode? Or keep the rewrites but compile to python still.

Reply:
With `mode=Mode("py")` the tests run a little faster than with `mode=FAST_COMPILE`.
I'll push this change now...

Reply:
Strange, is the failing Jax test possibly related?

Reply:
Multiple tests from the pytest workflows also failed.
I just switched to `FAST_COMPILE` let's see if it's faster or slower than the successful CI run of the commit before.